bigtable_rs/src/bigtable/read_rows.rs

```diff
@@ -33,6 +35,35 @@ pub async fn decode_read_rows_response(
     Ok(rows)
 }
 
+/// Decode the stream of `ReadRowsResponse` into a stream of `Result<(RowKey, Vec<RowCell>)>>`.
+pub async fn decode_read_rows_response_stream<'a>(
+    timeout: &'a Option<Duration>,
+    mut rrr: Streaming<ReadRowsResponse>,
+) -> Pin<Box<dyn Stream<Item = Result<(RowKey, Vec<RowCell>)>> + Send + 'a>> {
+    let started = Instant::now();
+
+    Box::pin(async_stream::stream! {
+        while let Ok(message) = rrr.message().await {
```

Comment:
Just curious, could this while block be replaced by some map block such as
https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.map ?

Reply:
I ended up using map. After removing timeout I was able to use 'static lifetime which makes things much cleaner.